### PR TITLE
Support Webpack 4

### LINKED ISSUE #59: Support webpack 4
Add webpack 4 to tests and update peerDependency.

### PR DESCRIPTION
**Pull request by fiznool**
Includes specs to test against Webpack v4.

Fixes #59

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -127,7 +91,10 @@ test('times out with error (DEPRECATED timeout: 1)', function (done) {
 })
 
 test('fails when both timeout and timeoutMs are set', function (done) {
+  // TODO this spec is causing jest not to clean up properly, we get a console warning
+  // 'Jest did not exit one second after the test run has completed.'
   compile2({ file: 'sleep.js.erb', timeout: 1, timeoutMs: 1000 }, done, function (stats) {
+    console.log(stats.compilation.errors)
```
Comment (PikachuEXE) [inline test.js]: Is this required?

**Comment by PikachuEXE** [comment]
Seem good for other things, but I am not a webpack expert :P

**Comment by PikachuEXE** [comment]
Maybe need to modify `.travis.yml` and add `TRAVIS_WEBPACK_VERSION="4"`
To actually test against webpack 4

**Comment by rhys-vdw** [comment]
Cheers @PikachuEXE, I was overseas and unable to reply.

Thanks for your contribution @fiznool. Would you be able to address the points that @PikachuEXE has raised and we can get this merged.

**Comment by PikachuEXE** [comment]
@fiznool 
Any update?

**Comment by kritik** [comment]
Didn't help me. Still getting an error on webpacker 4.2.2:
```
ERROR in ./app/javascript/packs/checkout.js.erb
Module build failed (from ./node_modules/rails-erb-loader/index.js):
Error: rails-erb-loader failed with code: 1
    at ChildProcess.<anonymous> (.../node_modules/rails-erb-loader/index.js:128:16)
    at ChildProcess.emit (events.js:203:13)
    at maybeClose (internal/child_process.js:1021:16)
    at Socket.<anonymous> (internal/child_process.js:430:11)
    at Socket.emit (events.js:203:13)
    at Pipe.<anonymous> (net.js:588:12)
````

**Comment by victor95pc** [comment]
Hi, guys any reason to not deploy it?
@PikachuEXE

**Comment by PikachuEXE** [comment]
@victor95pc 
See my last raised issue
Also even though I can merge and fix the small issue
I don't have release permission

**Comment by n-rodriguez** [comment]
Hi there! Any news?

**Comment by fiznool** [comment]
Sorry for the delay. I've pushed a new commit with the requested changes and also updated the travis config to test against node 10 and 12.

**Comment by fiznool** [comment]
Also fixes #80

**Comment by n-rodriguez** [comment]
> Also even though I can merge and fix the small issue
> I don't have release permission

yes but merge is the first step before release so let's do it :rocket:

**Comment by PikachuEXE** [comment]
I don't have release perm so you will have to use `master` until new version released

**Pull request merged**